### PR TITLE
Performance improvement for add team user when org has more than 1000 repositories

### LINKED ISSUE #19216: Slow performance (30+ seconds) when adding user to a group in large organization
### Description

Slow performance (7 minutes) on UI/API when adding/removing team in large organization (2000+ repos)

Steps to reproduce
1. Create organization with 2000+ repos
2. Create new team with:
* Repository access = All repos
* Permission = General access
* Access to sections = all to Read
* External Wiki/Issues = disabled
3. Add new user to this team
4. Wait 30-35 seconds until it done

### Gitea Version

1.16.3

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.21.0

### Operating System

CentOS Linux 7 (Core)

### How are you running Gitea?

APP_NAME = -ERASED-
RUN_USER = git
RUN_MODE = prod

[security]
INTERNAL_TOKEN     = -ERASED-
INSTALL_LOCK       = true
SECRET_KEY         = -ERASED-
PASSWORD_HASH_ALGO = pbkdf2
DISABLE_GIT_HOOKS  = false

[database]
DB_TYPE  = postgres
HOST     = -ERASED-
NAME     = -ERASED-
USER     = -ERASED-
PASSWD   = -ERASED-
SCHEMA   =
SSL_MODE = disable
CHARSET  = utf8
PATH     = /var/lib/gitea/data/gitea.db
LOG_SQL  = false

[repository]
ROOT = /var/lib/gitea/data/gitea-repositories

[server]
SSH_DOMAIN       = -ERASED-
DOMAIN           = -ERASED-
HTTP_PORT        = 3000
ROOT_URL         = -ERASED-
DISABLE_SSH      = false
SSH_PORT         = 22
LFS_START_SERVER = true
LFS_JWT_SECRET   = -ERASED-
OFFLINE_MODE     = false

[lfs]
PATH = /var/lib/gitea/data/lfs

[mailer]
ENABLED = true
HOST    = -ERASED-
FROM    = -ERASED-
USER    =
PASSWD  =

[service]
REGISTER_EMAIL_CONFIRM            = false
ENABLE_NOTIFY_MAIL                = true
DISABLE_REGISTRATION              = true
ALLOW_ONLY_EXTERNAL_REGISTRATION  = false
ENABLE_CAPTCHA                    = false
REQUIRE_SIGNIN_VIEW               = true
DEFAULT_KEEP_EMAIL_PRIVATE        = false
DEFAULT_ALLOW_CREATE_ORGANIZATION = false
ENABLE_TIMETRACKING               = false
DEFAULT_ENABLE_TIMETRACKING       = false
NO_REPLY_ADDRESS                  = noreply.localhost

[picture]
DISABLE_GRAVATAR        = false
ENABLE_FEDERATED_AVATAR = true

[openid]
ENABLE_OPENID_SIGNIN = false
ENABLE_OPENID_SIGNUP = false

[session]
PROVIDER = file

[log]
MODE      = file,console
LEVEL     = info
ROOT_PATH = /var/lib/gitea/log
ROUTER    = console,file

[metrics]
ENABLED = true

[mirror]
MIN_INTERVAL = 1m

[indexer]
ISSUE_INDEXER_TYPE = elasticsearch
ISSUE_INDEXER_CONN_STR = http://-ERASED-
ISSUE_INDEXER_NAME = gitea_issues
ISSUE_INDEXER_PATH = indexers/issues.bleve
REPO_INDEXER_ENABLED = true
REPO_INDEXER_TYPE = elasticsearch
REPO_INDEXER_CONN_STR = http://-ERASED-
REPO_INDEXER_NAME = gitea_codes
REPO_INDEXER_PATH = indexers/repos.bleve
UPDATE_BUFFER_LEN = 20
MAX_FILE_SIZE = 1048576
REPO_INDEXER_INCLUDE =
REPO_INDEXER_EXCLUDE = resources/bin/**
STARTUP_TIMEOUT = 30s

[git.timeout]
MIGRATE = 1800
MIRROR = 600

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by lunny**
Fix #19216

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Review by Gusted — APPROVED** [review]

**Comment by 6543** [comment]
@lunny I think we can merge this ... so we have at least a better  addUserToTeam.... but long term the crafted builder.Conditions should be returned by a func that can be used everywhere ....

tldr we should focus on making more builder constrains and use them instead of filter in for loops at go code ...

**Comment by lunny** [comment]
make L-G-T-M work.
